modules/reindex/src/main/java/org/opensearch/index/reindex/spi/RemoteReindexExtension.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.reindex.spi;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.index.reindex.BulkByScrollResponse;
+import org.opensearch.index.reindex.ReindexRequest;
+
+public interface RemoteReindexExtension {
+    /**
+     * Get an InterceptorProvider.
+     * @return ReindexRestInterceptorProvider implementation.
+     */
+    ReindexRestInterceptorProvider getInterceptorProvider();
+
+    /**
+     * Get a wrapper of ActionListener which is used to collect remote reindex metrics
+     * @return ActionListener wrapper implementation.
```

Comment:
Should we keep the comment more generic, as action listeners could serve more purpose than just the metrics.

Reply:
Made the comment generic